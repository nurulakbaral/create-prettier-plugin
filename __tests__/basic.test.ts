import { test, describe, expect } from 'vitest'
import { getFormattedSourceCode, getFixturesFolderName } from './utils'

const fixturesFolderName = getFixturesFolderName('./fixtures')
describe('JavaScript', () => {
  for (let folderName of fixturesFolderName) {
    if (folderName?.startsWith('javascript')) {
      test(folderName, async () => {
        let { actual, expected } = await getFormattedSourceCode(folderName, 'js')
        expect(actual, 'Missing actual (input) file').to.not.be.undefined
        expect(expected, 'Missing expected (output) file').to.not.be.undefined
        expect(actual).toBe(expected)
        expect(actual).toMatchSnapshot()
      })
    }
  }
})
describe('JSX', () => {
  for (let folderName of fixturesFolderName) {
    if (folderName?.startsWith('jsx')) {
      test(folderName, async () => {
        let { actual, expected } = await getFormattedSourceCode(folderName, 'jsx')
        expect(actual, 'Missing actual (input) file').to.not.be.undefined
        expect(expected, 'Missing expected (output) file').to.not.be.undefined
        expect(actual).toBe(expected)
        expect(actual).toMatchSnapshot()
      })
    }
  }
})
describe('TypeScript', () => {
  for (let folderName of fixturesFolderName) {
    if (folderName?.startsWith('typescript')) {
      test(folderName, async () => {
        let { actual, expected } = await getFormattedSourceCode(folderName, 'ts')
        expect(actual, 'Missing actual (input) file').to.not.be.undefined
        expect(expected, 'Missing expected (output) file').to.not.be.undefined
        expect(actual).toBe(expected)
        expect(actual).toMatchSnapshot()
      })
    }
  }
})
describe('TSX', () => {
  for (let folderName of fixturesFolderName) {
    if (folderName?.startsWith('tsx')) {
      test(folderName, async () => {
        let { actual, expected } = await getFormattedSourceCode(folderName, 'tsx')
        expect(actual, 'Missing actual (input) file').to.not.be.undefined
        expect(expected, 'Missing expected (output) file').to.not.be.undefined
        expect(actual).toBe(expected)
        expect(actual).toMatchSnapshot()
      })
    }
  }
})
