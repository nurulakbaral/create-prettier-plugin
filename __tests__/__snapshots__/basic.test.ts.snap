// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic Tests > javascript 1`] = `
"export function App() {
  return 'App'
}
"
`;

exports[`Basic Tests > jsx 1`] = `
"export function App() {
  return (
    <div>
      <h1>Hello</h1>
    </div>
  )
}
"
`;

exports[`Basic Tests > tsx 1`] = `
"import * as React from 'react'

export interface TAppProps {
  children: React.ReactNode
}

export function App({ children }: TAppProps) {
  return (
    <div>
      <h1>{children}</h1>
    </div>
  )
}
"
`;

exports[`Basic Tests > typescript 1`] = `
"interface TAppProps {
  name: string
}

export function App(props: TAppProps) {
  return props.name
}
"
`;
